src/composeappmanager.cc

```diff
@@ -475,23 +475,36 @@ data::InstallationResult ComposeAppManager::finalizeInstall(const Uptane::Target
     }
     // "finalize" (run) Apps that were pulled and created before reboot
     for (const auto& app_pair : getApps(target)) {
+      //Here the installation of the apps should be checked, and if they are not installed, they should be
```

Comment:
Most likely, you're correct, we should check if all target apps are pulled at this stage of the OTA update process and if not then pull them. Also, we may need to make it configurable, since this logic won't work for all use-cases we need to support (e.g. if a device needs to operate at a fully offline mode after reboot caused by an update).